src/utils.js

```diff
@@ -768,6 +768,8 @@ function getCompileFn(loaderContext, implementation, apiType) {
               webpackCompiler.hooks.shutdown.tap("sass-loader", () => {
                 compiler.dispose();
               });
+            } else {
+              compiler.dispose();
```

Comment:
I'm not a fan of if/elses deeply nested, but we can't return early in this scope, so it's okay.